kernel/multicore_bringup/src/aarch64.rs

```diff
@@ -64,56 +84,8 @@ pub fn handle_ap_cores(
-        // mapping is identity-mapped allows this code to enable paging
-        // transparently without messing up with the PC register.
-        //
-        // However, there is currently no way in Theseus to dynamically find a
-        // memory range that is identity mappable, so the address is hardcoded.
-        //
-        // Another solution was implemented instead, which is why this one is
-        // soft-commented. The CPU has instruction caches, so when paging gets
-        // enabled, it can still execute some instructions located near PC without
-        // using the page directory. This allows us to branch to a physical address,
-        // enable paging, and then branch to a virtual address (even though at the
-        // time of the second branch, PC has an invalid address) where the
-        // initialization can continue.
-        // Note: caches must not be flushed before this second branch.
-        //
-        // It would still be preferable to use the first solution, as it doesn't
-        // rely on a cache trick.
-        if false {
-            const ID_MAPPED_ENTRY_POINT: usize = 0x5001_0000;
-
-            entry_point_phys_addr = PhysicalAddress::new_canonical(ID_MAPPED_ENTRY_POINT);
-
-            // When the AArch64 core will enter startup code, it will do so with MMU disabled,
-            // which means these frames MUST be identity mapped.
-            let ap_startup_frames = memory::allocate_frames_by_bytes_at(entry_point_phys_addr, PAGE_SIZE)
-                .map_err(|_e| "handle_ap_cores(): failed to allocate AP startup frames")?;
-            let ap_startup_pages  = memory::allocate_pages_at(VirtualAddress::new_canonical(ID_MAPPED_ENTRY_POINT), ap_startup_frames.size_in_frames())
-                .map_err(|_e| "handle_ap_cores(): failed to allocate AP startup pages")?;
-
-            // map this RWX
-            let flags = PteFlags::new().valid(true).writable(true).executable(true);
-            ap_startup_mapped_pages = page_table.map_allocated_pages_to(
-                ap_startup_pages,
-                ap_startup_frames,
-                flags,
-            )?;
-
-            // copy the entry point code
-            // instructions must have an alignment of four bytes; a page's alignment will always be larger.
-            let dst = ap_startup_mapped_pages.as_slice_mut(0, PAGE_SIZE).unwrap();
-            let src = unsafe { (ap_entry_point as *const [u8; PAGE_SIZE]).as_ref() }.unwrap();
-            dst.copy_from_slice(src);
-
-        } else {
-            // get the physical address of ap_entry_point
-            let entry_point_virt_addr = VirtualAddress::new_canonical(ap_entry_point as usize);
-            entry_point_phys_addr = page_table.translate(entry_point_virt_addr).unwrap();
-        }
+        // get physical address of generated entry point
+        entry_point_phys_addr = page_table.translate(virt_addr).unwrap();
```

Comment:
am i missing something, or is this the identity mapping that we created above? If so, we don't need to translate it, we already know the physical address -- it's the same as the virtual one.

Reply:
I figured this would be cleaner than a cast but I can comment this and write a cast instead